edgedb-client/src/credentials.rs

```diff
@@ -19,13 +31,62 @@ pub struct Credentials {
     #[serde(default, skip_serializing_if="Option::is_none")]
     pub tls_cert_data: Option<String>,
     #[serde(default, skip_serializing_if="Option::is_none")]
-    pub tls_verify_hostname: Option<bool>,
+    tls_verify_hostname: Option<bool>, // deprecated
+    pub tls_security: TlsSecurity,
+    #[serde(skip)]
+    pub(crate) file_outdated: bool,
+}
+
+
+#[derive(Deserialize)]
```

Comment:
If you only deserialize this, you don't need `Cow`, just deserialize into strings.